src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts

```diff
@@ -49,7 +49,7 @@ export const useWorkspaceForm = ({
     Array<Pick<WorkspacePermissionSetting, 'id'> & Partial<WorkspacePermissionSetting>>
   >(initialPermissionSettingsRef.current);
 
-  const [selectedDataSources, setSelectedDataSources] = useState<DataSource[]>(
+  const [selectedDataSources, setSelectedDataSources] = useState<DataSourceConnection[]>(
```

Comment:
Can we keep using `DataSource[]` as data type here?

Reply:
`DataSourceConnection[]` was used in https://github.com/yubonluo/OpenSearch-Dashboards/tree/2.17/support-DQC

Reply:
Can't find related changes in that branch, can you paste me the code line by URL?

Reply:
https://github.com/yubonluo/OpenSearch-Dashboards/blob/679d689262ba5ac9f792d451963aadd29bfb846f/src/plugins/workspace/common/types.ts#L21

Reply:
My apologies, I made a mistake. workspace_form would get this `setSelectedDataSources` and pass it to SelectDataSourcePanel. SelectDataSourcePanel would need it https://github.com/Kapian1234/OpenSearch-Dashboards/blob/f22c7ac958ec5835c44a45e3447bd32d6b29b45b/src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx#L29